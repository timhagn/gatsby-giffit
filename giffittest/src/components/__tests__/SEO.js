import React from 'react'
import { render } from '@testing-library/react'
import { StaticQuery, useStaticQuery } from 'gatsby'
import SEOPage from '../SEO'

// Necessary to fixate generated className.
jest.mock('short-uuid')

const mockMetadata = {
  site: {
    siteMetadata: {
      title: `gbitest`,
      description: `A simple starter to show what gatsby-background-image can do.`,
      author: `Tim Hagn`,
    },
  },
}

describe('SEO Page', () => {
  beforeEach(() => {
    StaticQuery.mockImplementation(({ render }) => render({ ...mockMetadata }))
    useStaticQuery.mockImplementation(() => ({
      ...mockMetadata,
    }))
    // Freeze generated className.
    const uuid = require('short-uuid')
    uuid.generate.mockImplementation(() => '73WakrfVbNJBaAmhQtEeDv')
  })

  it('renders correctly', () => {
    const { container } = render(<SEOPage title={'test'} />)
    expect(container).toMatchSnapshot()
  })
})
