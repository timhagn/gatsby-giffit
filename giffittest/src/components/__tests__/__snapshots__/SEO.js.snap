// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO Page renders correctly 1`] = `<div />`;
